headers/data/tvector.hpp

```diff
@@ -0,0 +1,206 @@
+    return ( mapdata.find( round(t) ) != mapdata.end() );
+}
+
+//-----------------------------------------------------------------------
+/*! \brief 'Get' for time vector data type.
+ *
+ *  Return data associated with time t.
+ *  If no value exists, raise an exception.
+ */
+template <class T_data>
+const T_data &tvector<T_data>::get( double t ) const throw( h_exception ) {
+    typename std::map<double,T_data>::const_iterator itr = mapdata.find( round(t) );
+    if( itr != mapdata.end() )
+        return (*itr).second;
+    else {
+        Logger& glog = Logger::getGlobalLogger();
+        H_LOG( glog, Logger::SEVERE) << "No data at requested time= " << round(t) << std::endl;
+        H_THROW( "tvector: invalid time requested." );
+    }
+}
+
+/*! \brief mutable get method
+ *
+ * If the time vector object is not const, then get() returns a
+ * non-const reference, allowing an object to be modified in place.
+ */
+template <class T_data>
+T_data &tvector<T_data>::get( double t ) throw( h_exception ) {
+    typename std::map<double,T_data>::iterator itr = mapdata.find( round(t) );
+    if( itr != mapdata.end() )
+        return itr->second;
+    else {
+        Logger& glog = Logger::getGlobalLogger();
+        H_LOG( glog, Logger::SEVERE) << "No data at requested time= " << round(t) << std::endl; 
+        H_THROW( "tvector: invalid time requested" );
+    }
+}   
+
+template <class T_data>
+T_data &tvector<T_data>::operator[](double t) {
+    try {
+        return get(t);
+    }
+    catch(h_exception &fail) {
+        // h_exception is thrown when the requested object doesn't exist.
+        T_data newdat = T_data();
+        set(t, newdat);
+        return get(t);          // won't fail b/c we just created it
+    }
+}
+
+
+//-----------------------------------------------------------------------
+/*! \brief Return index of first element in vector.
+ *
+ *  Return index of first element in vector.
+ */
+template <class T_data>
+double tvector<T_data>::firstdate() const {
+    H_ASSERT( !mapdata.empty(), "no mapdata" );
+    return (*mapdata.begin()).first;
+}
+
+//-----------------------------------------------------------------------
+/*! \brief Return index of last element in vector.
+ *
+ *  Return index of last element in vector.
+ */
+template <class T_data>
+double tvector<T_data>::lastdate() const {
+    H_ASSERT( !mapdata.empty(), "no mapdata" );
+    return (*mapdata.rbegin()).first;
+}
+
+//-----------------------------------------------------------------------
+/*! \brief Return size of vector.
+ *
+ *  Return size of vector.
+ */
+template <class T_data>
+int tvector<T_data>::size() const {
+    return int( mapdata.size() );
+}
+
+/*! \brief truncate a time vector
+ *
+ *  \details The default is to wipe all of the data in the time vector
+ *           after the input date.  By setting the optional after
+ *           argument to false, you can wipe data before the input
+ *           date instead. 
```

Comment:
Maybe add a note describing why this is necessary, i.e. when `reset` then we truncate so that...

Reply:
FWIW, I cribbed the truncate method, the `after` argument, and the description you quoted from the `tseries` class.  I wound up using it in the reset methods, but the original purpose was to make `tvector` and `tseries` work the same way as much as possible.